src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -259,4 +260,178 @@ public void systemEndsPatientVisit() {
 		assertNull(visitsDashboardPage.getActiveVisit());
 		dashboardPage = visitsDashboardPage.goToPatientDashboard();
 	}
+
+	// User story: Adding New Allergy
+	@When("user clicks on Allergies link from Patient dashboard")
+	public void launchManageAllergies() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("system loads Allergies table page")
+	public void systemLoadsManageAllergies() {
+		assertEquals(getElement(patientHeaderId).getText(),
+				getElement(patientHeaderId).getText());
+		assertTrue(textExists("Allergies"));
+	}
+
+	@When("a user clicks Add New Allergy button")
+	public void addNewAllergy() {
+		allergyPage.clickOnAddNewAllergy();
+	}
+
+	@Then("the system loads Add New Allergy dashboard")
+	public void systemLoadsAddNewAllergy() {
+		assertTrue(textExists("Add New Allergy"));
+	}
+
+	@And("the user selects a new Allergy")
+	public void selectNewAllergy() {
+		allergyPage.clickOn(By.id("allergen-73667-label"));// Codeine
+	}
+
+	@And("the user clicks on the Save button")
+	public void saveAllergy() {
+		allergyPage.clickOn(By.id("addAllergyBtn"));
+		allergyPage.waitForPage();
+	}
+
+	@Then("the system adds a new Allergy")
+	public void systemAddsAllergy() {
+		assertTrue(textExists("Codeine"));
+	}
+
+	// User story: Add New Condition
+	@When("a user clicks on Conditions on Patient dashboard")
+	public void loadManageConditions() {
+		conditionsPage = (ConditionsPage) dashboardPage
+				.clickOnConditionsWidgetLink().waitForPage();
+	}
+
+	@Then("system loads Conditions Page")
+	public void loadsManageConditionsPage() {
+		assertTrue(conditionsPage.containsText("Conditions"));
+	}
+
+	@When("user clicks on Add new condition button")
+	public void clicksAddNewCondition() {
+		conditionPage = (ConditionPage) conditionsPage.clickOnAddNewCondition()
+				.waitForPage();
+	}
+
+	@And("user enters patient condition")
+	public void userEntersExistingCondition() {
+		conditionPage.typeInCondition(CONDITION);
+	}
+
+	@And("user clicks on save button")
+	public void clicksSaveCondition() {
+		conditionPage.clickSave();
+	}
+
+	@Then("system adds New Condition in Conditions table")
+	public void systemAddsNewCondition() {
+		assertNotNull(conditionsPage.getConditionsList());
+		dashboardPage = conditionsPage.clickReturn();
+	}
+
+	// User story: Adding and deleting New Allergy2
+	@When("a user clicks on Allergy2 link from Patient dashboard page2")
+	public void loadManageAllergy2() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("the system loads Allergy2 board page")
+	public void loadsManageAllergy2() {
+		assertEquals(getElement(patientHeaderId).getText(),
+				getElement(patientHeaderId).getText());
```

Comment:
Do we need the `assertEquals`?

Reply:
@Ndacyayisenga-droid  did you get to see the comment above?

Reply:
Did you have a chance to see the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/137/files#r717931652)?